probe-rs/src/probe/cmsisdap/mod.rs

```diff
@@ -717,6 +720,14 @@ impl DebugProbe for CmsisDap {
         Ok(speed_khz)
     }
 
+    fn set_scan_chain(&mut self, scan_chain: Vec<ScanChainElement>) -> Result<(), DebugProbeError> {
+        if self.active_protocol() != Some(WireProtocol::Jtag) {
+            tracing::warn!("Setting Scan Chain is only supported in JTAG mode");
+        }
+        self.scan_chain = Some(scan_chain);
```

Comment:
Maybe this method should also make a log that the scan chain is being set (or even better the location where it is actually configured), so when people fuck up the scanchain and probe-rs explodes weirdly later on, they get a hint :)

Reply:
Definitely. Even better, I should leverage `DebugProbeError` here and propagate the error up so it doesn't just silently explode as you say!